dev/integration_tests/android_views/lib/main.dart

```diff
@@ -0,0 +1,297 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:path_provider/path_provider.dart';
+
+import 'motion_event_diff.dart';
+
+MethodChannel channel = const MethodChannel('android_views_integration');
+
+const String kEventsFileName = 'touchEvents';
+
+/// Wraps a flutter driver [DataHandler] with one that waits until a delegate is set.
+///
+/// This allows the driver test to call [FlutterDriver.requestData] before the handler was
+/// set by the app in which case the requestData call will only complete once the app is ready
+/// for it.
+class FutureDataHandler {
+  final Completer<DataHandler> handlerCompleter = new Completer<DataHandler>();
+
+  Future<String> handleMessage(String message) async {
+    final DataHandler handler = await handlerCompleter.future;
+    return handler(message);
+  }
+}
+
+FutureDataHandler driverDataHandler = new FutureDataHandler();
+
+void main() {
+  enableFlutterDriverExtension(handler: driverDataHandler.handleMessage);
+  runApp(new MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Android Views Integration Test',
+      home: new Scaffold(
+        body: new PlatformViewPage(),
+      ),
+    );
+  }
+}
+
+class PlatformViewPage extends StatefulWidget {
+  @override
+  State createState() => new PlatformViewState();
+}
+
+class PlatformViewState extends State<PlatformViewPage> {
+  static const int kEventsBufferSize = 1000;
+
+  MethodChannel viewChannel;
+
+  /// The list of motion events that were passed to the FlutterView.
+  List<Map<String, dynamic>> flutterViewEvents = <Map<String, dynamic>>[];
+
+  /// The list of motion events that were passed to the embedded view.
+  List<Map<String, dynamic>> embeddedViewEvents = <Map<String, dynamic>>[];
+
+  @override
+  Widget build(BuildContext context) {
+    return new Column(
+      children: <Widget>[
+        new SizedBox(
+            height: 300.0,
+            child: new AndroidView(
```

Comment:
Feel free to ignore or do in a different PR: I wonder if we can integrate a 'performance test' in here. Besides the synthesized down time and event times, can we wrap this whole thing in a transparent gesture detector, record the actual time here and put that into a list as well and also record the time on the native size, pass it back along with the motionevents and make sure the time delta doesn't get past a threshold?

Reply:
I plan to add a benchmark for the roundtrip of MotionEvent->PointerEvents->MotionEvent using this integration test.